aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/FactoryExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace BlazorServerDbContextExample.Data
+{
+    /// <summary>
+    /// Make it easier to register the factory.
+    /// </summary>
+    public static class FactoryExtensions
+    {
+        /// <summary>
+        /// Registers the <seealso cref="DbContextFactory{TContext}"/> with DI.
+        /// </summary>
+        /// <typeparam name="TContext">The instance of <see cref="DbContext"/> to register.</typeparam>
+        /// <param name="collection">The instance of the <see cref="IServiceCollection"/>.</param>
+        /// <param name="optionsAction">Optional access to the <see cref="DbContextOptions{TContext}"/>.</param>
+        /// <param name="contextAndOptionsLifetime">Set the <see cref="ServiceLifetime"/> of the factory and options.</param>
+        /// <returns>The registered <see cref="IServiceCollection"/>.</returns>
+        public static IServiceCollection AddDbContextFactory<TContext>(
+            this IServiceCollection collection,
+            Action<DbContextOptionsBuilder> optionsAction = null,
+            ServiceLifetime contextAndOptionsLifetime = ServiceLifetime.Singleton)
```

Comment:
I'm not sure the factory being a singleton will work. Unless it is able to create its own scope and dispose it at the right time, creating Disposable objects through the DI container will be problematic here.

Reply:
I'm not sure I understand. It is using activator utilities to create a new instance and return it, so there are no references to the created contexts in either the factory or the SP ... am I missing something?